client/question/domain/PrereqCheckErrorObjectFactory.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating a PrereqCheckError of a student's TIE
+ * session.
+ */
+
+tie.factory('PrereqCheckErrorObjectFactory', [
+  function() {
+    /**
+     * PrereqCheckError objects are used to store information related to Prereq
+     errors.
+     */
+
+    /**
+     * Constructor for PrereqCheckError.
+     *
+     * @constructor
+     */
+    var PrereqCheckError = function(errorName, errorLineNumber) {
```

Comment:
If this is just going to be scoped to wrong language errors then the domain object should be WrongLanguageCheckError or similar.

Alternatively if this is going to be used for other types of errors as well then perhaps we should have a this._errorType that's set to ERROR_TYPE_WRONG_LANGUAGE or null (for now). That would make it easier to add other error types in the future.